tools/analyse-storage/src/main.rs

```diff
@@ -0,0 +1,268 @@
+        "col38" => Some(DBCol::BlockMerkleTree),
+        "col39" => Some(DBCol::ChunkHashesByHeight),
+        "col40" => Some(DBCol::BlockOrdinal),
+        "col41" => Some(DBCol::_GCCount),
+        "col42" => Some(DBCol::OutcomeIds),
+        "col43" => Some(DBCol::_TransactionRefCount),
+        "col44" => Some(DBCol::ProcessedBlockHeights),
+        "col45" => Some(DBCol::Receipts),
+        "col46" => Some(DBCol::CachedContractCode),
+        "col47" => Some(DBCol::EpochValidatorInfo),
+        "col48" => Some(DBCol::HeaderHashesByHeight),
+        "col49" => Some(DBCol::StateChangesForSplitStates),
+        _ => {
+            for db_col in DBCol::iter() {
+                if format!("{}", db_col) == col {
+                    return Some(db_col);
+                }
+            }
+            return None;
+        }
+    }
+}
+
+fn get_all_col_familiy_names() -> Vec<DBCol> {
+    DBCol::iter().collect()
+}
+
+fn print_results(
+    sizes_count: &Vec<(usize, usize)>,
+    size_count_type: &str,
+    limit: usize,
+    total_num_of_pairs: usize,
+) {
+    println!(
+        "Total number of pairs read {}\n",
+        sizes_count.into_iter().map(|(_, count)| count).sum::<usize>()
+    );
+
+    // Print out distributions
+    println!("{} Size Distribution:", size_count_type);
+    println!(
+        "Minimum size {}: {:?}",
+        size_count_type,
+        sizes_count.iter().map(|(size, _)| size).min().unwrap()
+    );
+    println!(
+        "Maximum size {}: {:?}",
+        size_count_type,
+        sizes_count.iter().map(|(size, _)| size).max().unwrap()
+    );
+    println!("Most occuring size {}: {:?}", size_count_type, sizes_count.first().unwrap());
+    println!("Least occuring size {}: {:?}", size_count_type, sizes_count.last().unwrap());
+
+    let total_sizes_bytes_sum = sizes_count.iter().map(|a| a.0 * a.1).sum::<usize>();
+    println!(
+        "Average size {}: {:?}",
+        size_count_type,
+        total_sizes_bytes_sum as f64 / total_num_of_pairs as f64
+    );
+    let mut size_bytes_median = 0;
+    let mut median_index = total_num_of_pairs / 2;
+    for (size, count) in sizes_count.iter().take(limit) {
+        if median_index < *count {
+            size_bytes_median = *size;
+            break;
+        } else {
+            median_index -= count;
+        }
+    }
+    println!("Median size {} {}", size_count_type, size_bytes_median);
+    for (size, count) in sizes_count.iter().take(limit) {
+        println!("Size: {}, Count: {}", size, count);
+    }
+    println!("");
+}
+
+fn rocksdb_column_options(col: DBCol) -> Options {
+    let mut opts = Options::default();
+    opts.set_level_compaction_dynamic_level_bytes(true);
+
+    if col.is_rc() {
+        opts.set_merge_operator("refcount merge", RocksDB::refcount_merge, RocksDB::refcount_merge);
+        opts.set_compaction_filter("empty value filter", RocksDB::empty_value_compaction_filter);
+    }
+    opts
+}
+
+fn read_all_pairs(
```

Comment:
It is not clear from the context that it is not lexicographical sorting but sorting in non-decreasing occurrence order. Can we mention this somewhere, like a comment for this function?

Reply:
Ah yes of course, it is for getting histogram like view of key and value sizes